core/state_transition.go

```diff
@@ -185,41 +187,47 @@ func (st *StateTransition) useGas(amount uint64) error {
 	return nil
 }
 
-func (st *StateTransition) buyGas() error {
-	mgval := new(big.Int).Mul(new(big.Int).SetUint64(st.msg.Gas()), st.gasPrice)
+// payFees deducts gas and gateway fees from sender balance and returns error if the sender cannot pay.
```

Comment:
note that it also sets the initial gas amount

Reply:
Added a note about that in the comments.